### ISSUE TITLE
Tests should use a modern hash (SHA-256 instead of MD5)

### ISSUE THREAD

**Issue by johnhawkinson** [rg3/youtube-dl#12363]
`youtube-dl`'s download tests all use MD5 as a hash. While it's probably not cryptographically critical that download testing use a secure hash, the world has moved past MD5, and past SHA-1. `youtube-dl` should at least support SHA-256 in the `info_dict`, and use whichever hash has been supplied in the `_TEST`'s `info_dict`.

**Comment by yan12125** [comment]
Good idea. Just checked it, python 2.6 supports up to sha512. An implementation can be allowing sha256 (or 512) for new _TESTS as changing all tests at a time is impractical.

**Comment by rautamiekka** [comment]
SHA-512 would be better than 256, but even better would be using SHA3-512 even if it takes having an additional lib

But definitely SHA-256 **at minimum**.

**Comment by yan12125** [comment]
SHA3 is not available in stdlib until Python 3.6 http://bugs.python.org/issue16113. I don't think there's a need to bring another external dependency for tests. They don't need cryptographically secure hashes.
